testnet/stacks-node/src/event_dispatcher.rs

```diff
@@ -259,7 +251,7 @@ pub struct EventDispatcher {
     mempool_observers_lookup: HashSet<u16>,
     stx_observers_lookup: HashSet<u16>,
     any_event_observers_lookup: HashSet<u16>,
-    boot_receipts: Vec<StacksTransactionReceipt>,
+    boot_receipts: RefCell<Option<Vec<StacksTransactionReceipt>>>,
```

Comment:
Why use a RefCell here?

Reply:
Only using it for mutability, to be able to drop the (very large) array from memory once consumed. Is there a better way to handle this?

Reply:
I think you'd get the same behavior from just calling `.take()` on an Option.

Reply:
Oh, I see -- you're using the RefCell so that the event dispatcher doesn't need to be invoked mutably. Hmm, I think that's fine then.

Reply:
We're not worried about concurrent access here, are we?  Otherwise it might be prudent to use a `Arc` instead.

Reply:
Looking through the call graphs I don't think this would be an issue, not sure.. @kantai?

Reply:
Assuming it behaves the same as a `RefCell` (it should), why not make it a `Arc` to be safe?  I don't think access to this data is performance-critical, so we don't need to worry about the tiny extra cost of acquiring/releasing the internal mutex.

Reply:
I gave this a try and can't get the same mutability provided by `RefCell.borrow_mut()`. Any suggestions?

Reply:
Got this to work with an `Arc<Mutex<...>>` (thx @kantai for the guidance). @jcnelson does this look good to you?

Reply:
Almost -- can you replace the `.unwrap()` with an `.expect()` with an error message to indicate that the mutex got corrupted?  I think the panicking crash is the right thing to do either way, but I'd prefer a descriptive error message.

Reply:
Done!